core/client/fs/src/main/java/alluxio/client/block/StreamFactory.java

```diff
@@ -42,30 +47,17 @@ private StreamFactory() {} // prevent instantiation
    * @param options the out stream options
    * @return the {@link OutputStream} object
    */
-  public static BlockOutStream createLocalBlockOutStream(FileSystemContext context, long blockId,
+  public static BlockOutStream createBlockOutStream(FileSystemContext context, long blockId,
       long blockSize, WorkerNetAddress address, OutStreamOptions options) {
-    if (NettyUtils.isDomainSocketSupported(address)) {
+    if (address.getHost().equals(NetworkAddressUtils.getClientHostName()) && Configuration
```

Comment:
`WorkerNetAddress.getHost().equals(NetworkAddressUtils.getClientHostName())` should go to a util